src/ast/generated/gen-make-functions-script.ts

```
@@ -0,0 +1,301 @@
+                            t.identifier(namespace),
+                            typ.typeName,
+                        ),
+                    );
+                } else {
+                    // Leave the identifier unchanged, but check if it has type parameters
+                    const typRef = t.tsTypeReference(typ.typeName);
+                    if (typ.typeParameters) {
+                        typRef.typeParameters = t.tsTypeParameterInstantiation(
+                            typ.typeParameters.params.map((t) =>
+                                qualifyType(namespace, t, decls),
+                            ),
+                        );
+                    }
+                    return typRef;
+                }
+            }
+            // Leave the type as is
+            return typ;
+        }
+        case "TSUnionType": {
+            return t.tsUnionType(
+                typ.types.map((t) => qualifyType(namespace, t, decls)),
+            );
+        }
+        case "TSArrayType": {
+            return t.tsArrayType(
+                qualifyType(namespace, typ.elementType, decls),
+            );
+        }
+        case "TSTypeOperator": {
+            const op = t.tsTypeOperator(
+                qualifyType(namespace, typ.typeAnnotation, decls),
+            );
+            op.operator = typ.operator;
+            return op;
+        }
+        case "TSTupleType": {
+            if (
+                typ.elementTypes.every((t) => t.type !== "TSNamedTupleMember")
+            ) {
+                return t.tsTupleType(
+                    typ.elementTypes.map((t) =>
+                        qualifyType(namespace, t, decls),
+                    ),
+                );
+            } else {
+                // Currently unsupported
+                throw new Error(
+                    "TSNamedTupleMember is currently not supported in TSTupleType",
+                );
+            }
+        }
+
+        default:
+            return typ;
```

I would rather throw an exception here, because there might be (or will be) types with identifiers inside that are not (will not be) handled.

Throwing an exception here would be slightly safer.

---

Done